frontend/src/components/dashboard/settings/verify-custom-domain-modal/VerifyCustomDomainModal.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useContext } from 'react'
+import { PrimaryButton, ErrorBlock } from 'components/common'
+import EmailValidationInput from 'components/dashboard/create/email/EmailValidationInput'
+import { ModalContext } from 'contexts/modal.context'
+import { verifyFromAddress } from 'services/email.service'
+
+import ConfirmImage from 'assets/img/confirm-modal.svg'
+import FailureImage from 'assets/img/failure.png'
+import SuccessImage from 'assets/img/success.png'
+import styles from './VerifyCustomDomainModal.module.scss'
+
+enum VerifyEmailStep {
+  Verify,
+  Success,
+  Failure,
+}
+
+const VerifyCustomDomainModal = ({
+  label,
+  onSuccess,
+}: {
+  label: string
+  onSuccess: Function
+}) => {
+  const [step, setStep] = useState(VerifyEmailStep.Verify)
+  const [errorMessage, setErrorMessage] = useState('')
+  const modalContext = useContext(ModalContext)
+
+  async function verifyCustomDomain(recipient: string) {
+    setErrorMessage('')
+    try {
+      await verifyFromAddress(recipient, label)
+      setStep(VerifyEmailStep.Success)
+      onSuccess()
+    } catch (e) {
+      console.error(e)
+      setErrorMessage(e.message)
+      setStep(VerifyEmailStep.Failure)
+    }
+  }
+
+  function renderValidate() {
+    return (
+      <>
+        <img src={ConfirmImage} alt="" />
+        <h2>Verify From Address</h2>
+        <p>
+          To verify that emails can be sent from <b>{label}</b>, please enter an
+          available recipient to receive a validation message.
+        </p>
+        {
+          <EmailValidationInput
+            onClick={verifyCustomDomain}
+          ></EmailValidationInput>
+        }
+      </>
+    )
+  }
+
+  function renderVerifyCredStep() {
+    switch (step) {
+      case VerifyEmailStep.Verify:
+        return renderValidate()
+      case VerifyEmailStep.Success:
+        return (
+          <div className={styles.centerAlign}>
+            <img src={SuccessImage} alt="" />
+            <h3>
+              We sent your message from {label}. Check your inbox to see if you
+              received it.
+            </h3>
+            <PrimaryButton
+              className={styles.padTop}
+              onClick={() => modalContext.setModalContent(null)}
+            >
+              Done
+            </PrimaryButton>
+          </div>
+        )
+      // Credentials failed to store
+      case VerifyEmailStep.Failure:
+        return (
+          <div className={styles.centerAlign}>
+            <img src={FailureImage} alt="" />
+            <h3>Sorry, something went wrong.</h3>
+            <ErrorBlock>{errorMessage}</ErrorBlock>
```

Comment:
When verification fails, the error message is not very informative. It's not clear how they can remedy the verification failure. In this case, the cause could either be that the agency's IT has not added or has removed the necessary DNS records. 

Perhaps we can add a message to tell them to try again in a while (while their IT team add the records) or to contact us. 

<img width="804" alt="94486472-103c8280-0212-11eb-8007-3accced4b8dc" src="https://user-images.githubusercontent.com/3666479/94771250-6879ad00-03e9-11eb-9ae2-30cc66c1ae08.png">

Reply:
Will change in backend